framework/src/multiapps/MultiApp.C

```diff
@@ -517,6 +532,52 @@ MultiApp::fillPositions()
           _input_files.push_back(input_files[p_file_it]);
     }
   }
+  else if (isParamValid("positions_objects"))
+  {
+    std::vector<PositionsName> positions_param_objs =
+        getParam<std::vector<PositionsName>>("positions_objects");
+    std::vector<FileName> input_files = getParam<std::vector<FileName>>("input_files");
+
+    if (input_files.size() != 1 && positions_param_objs.size() != input_files.size())
+      mooseError("Number of input_files for MultiApp ",
+                 name(),
+                 " must either be only one or match the number of positions_objects specified");
+
+    // Clear out the _input_files because we're going to rebuild it
+    if (input_files.size() != 1)
+      _input_files.clear();
+
+    // Keeps track of where each positions object start in terms of subapp numbers
+    unsigned int offset = 0;
+
+    for (unsigned int p_obj_it = 0; p_obj_it < positions_param_objs.size(); p_obj_it++)
+    {
+      const std::string positions_name = positions_param_objs[p_obj_it];
+      auto positions_obj = &_fe_problem.getPositionsObject(positions_name);
+
+      auto data = positions_obj->getPositions(true);
+
+      for (const auto & d : data)
+        _positions.push_back(d);
+
+      // Save the number of positions for this input file
+      _npositions_inputfile.push_back(data.size());
+
+      if (!positions_obj)
+        paramError("positions_objects",
+                   "'" + positions_name + "' is not of the expected type. Should be a Positions");
+
+      // Keep track of which positions is tied to what subapp
+      for (unsigned int i = 0; i < data.size(); ++i)
```

Comment:
```suggestion
      for (const auto i : index_range(data))
```

Reply:
keeping old one.
with new one, i is reported as unused and would need some trick to bypass that